src/integrations/mailerlite/class-mailerlite-client.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Mailerlite client implementation
+ *
+ * @package Eightshift_Forms\Integrations\Mailerlite
+ */
+
+declare( strict_types=1 );
+
+namespace Eightshift_Forms\Integrations\Mailerlite;
+
+use Eightshift_Forms\Hooks\Filters;
+use \MailerLiteApi\MailerLite as MailerLiteClient;
+use \GuzzleHttp\Client as GuzzleHttp;
+use \Http\Adapter\Guzzle6\Client as Guzzle6;
+
+/**
+ * Mailerlite integration class.
+ */
+class Mailerlite_Client implements Mailerlite_Client_Interface {
+
+  /**
+   * Mailerlite client object.
+   *
+   * @var MailerLiteClient
+   */
+  protected $client;
+
+  /**
+   * Sets the config because we can't set config during construction (filters aren't yet registered)
+   *
+   * @return void
+   */
+  public function set_config() {
+    $api_key = \has_filter( Filters::MAILERLITE ) ? \apply_filters( Filters::MAILERLITE, 'api_key' ) : '';
+
+    $guzzle = new GuzzleHttp();
+    $guzzleClient = new Guzzle6($guzzle);
+
+    $this->client = new MailerLiteClient($api_key, $guzzleClient);
```

Comment:
Why is there a direct call towards these clients here? Shouldn't this be abstracted out and added using DI?

Reply:
This is because of this specific implementation only on this integration.
https://github.com/mailerlite/mailerlite-api-v2-php-sdk#use-your-preferred-http-client

I didn't see the point of extracting this into abstraction